Code/Tools/SerializeContextTools/main.cpp

```diff
@@ -102,61 +103,79 @@ int main(int argc, char** argv)
 {
     using namespace AZ::SerializeContextTools;
 
-    bool result = false;
-    Application application(argc, argv);
-    // Direct Raw Debug Trace Messages to stderr
-    if (auto console = AZ::Interface<AZ::IConsole>::Get(); console != nullptr)
+    constexpr int StdoutDescriptor = 1;
+    AZ::IO::FileDescriptorCapturer stdoutCapturer(StdoutDescriptor);
+
+    // Send stdout output to stderr if the executed command returned a failure
+    bool suppressStderr = false;
+    auto SendStdoutToError = [&suppressStderr](AZStd::span<AZStd::byte const> outputBytes)
     {
-        constexpr auto traceRedirectStream = AZ::Debug::RedirectCStream::Stderr;
-        console->PerformCommand("bg_redirectrawoutput",
-            { AZ::CVarFixedString::format("%d", static_cast<int>(traceRedirectStream)) });
-    }
+        if (!suppressStderr)
+        {
+            constexpr int StderrDescriptor = 2;
+            AZ::IO::PosixInternal::Write(StderrDescriptor, outputBytes.data(), aznumeric_cast<int>(outputBytes.size()));
+        }
+    };
+
+    stdoutCapturer.Start();
+    Application application(argc, argv, &stdoutCapturer);
     AZ::ComponentApplication::StartupParameters startupParameters;
     application.Start({}, startupParameters);
 
+    bool result = false;
     const AZ::CommandLine* commandLine = application.GetAzCommandLine();
-    if (commandLine->GetNumMiscValues() < 1)
-    {
-        PrintHelp();
-        result = true;
-    }
-    else
+    bool commandExecuted = false;
+    if (commandLine->GetNumMiscValues() >= 1)
     {
-        const AZStd::string& action = commandLine->GetMiscValue(0);
-        if (AZ::StringFunc::Equal("dumpfiles", action.c_str()))
+        AZStd::string_view action = commandLine->GetMiscValue(0);
```

Comment:
Might be easier to set `commandExecuted = true;` here at the top, then add an `else { commandExecuted = false; }` clause at the end.

Reply:
```
bool commandExecuted = false;
if (commandLine->GetNumMiscValues() >= 1)
{
    commandExecuted = true;
    if (AZ::StringFunc::Equal("dumpfiles", action))
    {
        result = Dumper::DumpFiles(application);
    }
    ...
    else
    {
        commandExecuted = false;
    }
}
```